pi5/board.go

```diff
@@ -206,94 +289,102 @@ func (b *pinctrlpi5) setPulls() {
 	}
 }
 
-type pinctrlpi5 struct {
-	resource.Named
-	mu sync.Mutex
-
-	gpioMappings map[string]gl.GPIOBoardMapping
-	logger       logging.Logger
-
-	gpios      map[string]*pinctrl.GPIOPin
-	interrupts map[string]*pinctrl.DigitalInterrupt
-
-	boardPinCtrl pinctrl.Pinctrl
-
-	cancelCtx               context.Context
-	cancelFunc              func()
-	activeBackgroundWorkers sync.WaitGroup
-
-	pulls map[int]byte // mapping of gpio pin to pull up/down
-}
-
 // AnalogByName returns the analog pin by the given name if it exists.
 func (b *pinctrlpi5) AnalogByName(name string) (board.Analog, error) {
 	return nil, errors.New("analogs not supported")
 }
 
-// DigitalInterruptByName returns the interrupt by the given name if it exists.
-func (b *pinctrlpi5) DigitalInterruptByName(name string) (board.DigitalInterrupt, error) {
-	b.mu.Lock()
-	defer b.mu.Unlock()
+// the implementation of digitalInterruptByName. The board mutex should be locked before calling this.
```

Comment:
why can't you lock the mutex inside the function?

Reply:
I had to move the mutex to outside of this function because I needed reconfigure to be able to create interrupts. 

since reconfigure also locks the mutex having the lock within this function would cause the board to lockup when configuring interrupts

Reply:
more context in [this comment](https://github.com/viam-modules/raspberry-pi/pull/53/files#r1878393228)